gustaf/helpers/data.py

```diff
@@ -0,0 +1,446 @@
+
+    Note, if you really really want, it is possible to change the tracked
+    array with out setting modified flag.
+    """
+
+    __slots__ = ("_modified", "_source")
+
+    def __array_finalize__(self, obj):
+        """Sets default flags for any arrays that maybe generated based on
+        tracked array."""
+        self._modified = True
+        self._source = int(0)
+
+        if isinstance(obj, type(self)):
+            if isinstance(obj._source, int):
+                self._source = obj
+            else:
+                self._source = obj._source
+
+    @property
+    def mutable(self):
+        return self.flags['WRITEABLE']
+
+    @mutable.setter
+    def mutable(self, value):
+        self.flags.writeable = value
+
+    def _set_modified(self):
+        """set modified flags to itself and to the source."""
+        self._modified = True
+        if isinstance(self._source, type(self)):
+            self._source._modified = True
+
+    def copy(self, *args, **kwargs):
+        """copy gives np.ndarray.
+
+        no more tracking.
+        """
+        return np.array(self, copy=True)
+
+    def view(self, *args, **kwargs):
+        """Set writeable flags to False for the view."""
+        v = super(self.__class__, self).view(*args, **kwargs)
+        v.flags.writeable = False
+        return v
+
+    def __iadd__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__iadd__(*args, **kwargs)
+
+    def __isub__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__isub__(*args, **kwargs)
+
+    def __imul__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__imul__(*args, **kwargs)
+
+    def __idiv__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__idiv__(*args, **kwargs)
+
+    def __itruediv__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__itruediv__(*args, **kwargs)
+
+    def __imatmul__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__imatmul__(*args, **kwargs)
+
+    def __ipow__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__ipow__(*args, **kwargs)
+
+    def __imod__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__imod__(*args, **kwargs)
+
+    def __ifloordiv__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__ifloordiv__(*args, **kwargs)
+
+    def __ilshift__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__ilshift__(*args, **kwargs)
+
+    def __irshift__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__irshift__(*args, **kwargs)
+
+    def __iand__(self, *args, **kwargs):
```

Comment:
I have to admit, this is far beyond my python expertise and I would have to google every single line in order to understand this. So I take this as is